### PR TITLE
[DOCS] Expand simple query string query's synonym section

### PR DESCRIPTION
**Pull request by jrodewig**
As noted in #68720, the simply query string query's `simple_query_string` works for all multi-position tokens, not just multi-word synonyms.

### Previews
- Multi-position tokens: https://elasticsearch_68753.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/token-graphs.html#token-graphs-multi-position-tokens
- Simple query string query: https://elasticsearch_68753.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/query-dsl-simple-query-string-query.html#simple-query-string-synonyms

### CODE REVIEW

File: docs/reference/query-dsl/simple-query-string-query.asciidoc [hunk 1]
```diff
@@ -273,33 +274,39 @@ GET /_search
...
-synonym_graph>> token filter. When this filter is used, the parser creates a phrase query for each multi-terms synonyms.
-For example, the following synonym: `"ny, new york"` would produce:
+The `simple_query_string` query supports
+<<token-graphs-multi-position-tokens,multi-position tokens>> produced by graph
+token filters, such as <<analysis-synonym-graph-tokenfilter,`synonym_graph`>> or
+<<analysis-word-delimiter-graph-tokenfilter,`word_delimiter_graph`>>.
+
+By default, the parser creates a <<query-dsl-match-query-phrase,`match_phrase`>>
+query for each multi-position token. For example, the parser creates a
+`match_phrase` query for the multi-word synonym `ny, new york`:
```
Comment (jrodewig) [inline docs/reference/query-dsl/simple-query-string-query.asciidoc:286]: I intentionally chose to leave this example largely as is. Synonyms seem to be a more common use case. However, I'm happy to change the example or add another if wanted.

File: docs/reference/query-dsl/simple-query-string-query.asciidoc [hunk 2]
```diff
@@ -273,33 +274,39 @@ GET /_search
...
-===== Synonyms
+===== Multi-position tokens
 
-The `simple_query_string` query supports multi-terms synonym expansion with the <<analysis-synonym-graph-tokenfilter,
-synonym_graph>> token filter. When this filter is used, the parser creates a phrase query for each multi-terms synonyms.
-For example, the following synonym: `"ny, new york"` would produce:
+The `simple_query_string` query supports
+<<token-graphs-multi-position-tokens,multi-position tokens>> produced by graph
+token filters, such as <<analysis-synonym-graph-tokenfilter,`synonym_graph`>> or
+<<analysis-word-delimiter-graph-tokenfilter,`word_delimiter_graph`>>.
```
Comment (imotov) [inline docs/reference/query-dsl/simple-query-string-query.asciidoc:282]: Maybe add some information about decompounders such as nori as well?

**Review by imotov — APPROVED** [review]
LGTM. But I think it would be great for somebody from the search team to look at it as well, since they are much closer to this code.

File: docs/reference/query-dsl/simple-query-string-query.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/query-dsl/simple-query-string-query.asciidoc]: Thanks @imotov. After looking at this again, I decided to remove the graph token filters reference. It's a bit redundant with the multi-position token link. I added a reference to the `nori_tokenizer` to those docs with 1cc2ab7310ce475ed292fb1a6f178986625fb2ed.

**Review by romseygeek — APPROVED** [review]
LGTM

**Comment by jrodewig** [comment]
Thanks @romseygeek @imotov

**Pull request merged**